src/infrastructure/managing-items.md

```diff
@@ -83,28 +83,28 @@ $item = $manager->create( [/*...*/] );
 This creates a new object with some default values for the item set. You can add properties if you pass a list of key/value pairs in the same structure as returned by toArray() of the item as first argument. The item doesn't have an unique ID by default and is only available in memory.
```

Comment:
This sentence "You can add..." is very confusing, I do not really get what it tries to explain. Any chance to improve this?

Reply:
Is "You can set ..." better?

Reply:
Sorry, I was unclear. I meant the whole sentence. For me, this is too much: "...if you pass a list of key/value pairs in the same structure as returned by toArray() of the item as first argument" 
I have no idea what the return value looks like, why a "toArray()" is suddenly mentioned, what its first argument is, why it is that exactly the first argument is refereneced (why not 2nd?), also what the first argument looks like? Maybe an example would help?
I assume eventually sth like `[ $key => $value, $key2 => $value2 ]` can be passed?

Reply:
Yes, and a concrete example for products would be
```
['product.code' => 'test', 'product.label' => 'Test']
```
to initialize the new (product) item.

Reply:
And what would the first key/value pair of
`var_export( $item->toArray() )` (if that is what is meant in this passage)
look like after initialisation (without having added properties yet)?

Reply:
If `['product.code' => 'test', 'product.label' => 'Test']` is passed as parameter to `create()` then the same is returned by `toArray()` again. If nothing is passed, emtpy values will be returned (`['product.code' => '', 'product.label' => '']`)

Reply:
I am sorry for being so stupid, but I still don't understand, why it is necessary to mention the "toArray()" thingy. Also, the docu says that a new item "some default values" is created, so I would expect that those default values would be returned, if I use "$item->toArray()", and those default values show the structure of how values can be passed to the `create()` function.

Which information would be wrong or missing, if this sentence was written like this:

"[...] You can add properties by passing a list of key/value pairs as first argument like this:

```php
$item = $manager->create(['product.code' => 'test', 'product.label' => 'Test']);
```

If you want to retrieve those values again, use `$item->toArray();`. [...]"

Currently I would assume I need to retrieve those values in the html client, but that is probably wrong, or at least not the only place. Maybe this is not important, since at this point a developer should know, how to use new value; but in case it is, it should be mentioned, IMO.

I also assume there is no way to retrieve those "default values" that are mentioned? Or is there? Do I need to know (as a developer) what those default values exactly are? Can they be displayed in the html templates?

Reply:
The `toArray()` was mentioned to give the user a hint how the keys must look like. In your example it will only work for product items and you now need to mention that for other items, the keys are different. Then, the users will ask how must they look like and this is the time that a reference to `toArray()` is handy.

Reply:
New commit added to this PR. Feedback most welcome!